src/libos/crates/host-socket/src/stream/mod.rs

```diff
@@ -185,6 +185,47 @@ impl<A: Addr, R: Runtime> StreamSocket<A, R> {
             .unregister_observer(observer)
             .ok_or_else(|| errno!(ENOENT, "the observer is not registered"))
     }
+
+    pub fn addr(&self) -> Result<A> {
+        let state = self.state.read().unwrap();
+        match &*state {
+            State::Init(init_stream) => init_stream.common().addr().map_or_else(
+                || init_stream.common().get_addr_from_host(),
+                |addr| Ok(addr),
+            ),
+            State::Connect(connecting_stream) => connecting_stream.common().addr().map_or_else(
+                || connecting_stream.common().get_addr_from_host(),
+                |addr| Ok(addr),
+            ),
```

Comment:
Could you shred some lights on why init or connecting sockets should call `get_addr_from_host`? For unix stream sockets, whose addresses are unnamed by default? (I don't if this should be the right behavior. Need some testing to confirm.) If so, I think the more correct logic is to `stream.common().addr().unwrap_or_default()`.

Reply:
> If the socket is not bound to an address, the call returns with the family set, and **the rest of the structure is set to 0**.

> The getsockname function is often used to find the port assigned to a socket after the socket has been implicitly bound to a port. For example, an application can call connect without previously calling bind. In this case, the connect function completes the binding necessary by assigning a port to the socket. This assignment can be found with a call to getsockname.

> A UNIX domain stream socket that is not bound to a path name by using the bind function is **unnamed**. Similarly, the two UNIX domain sockets that are created by using the socketpair function are **unnamed**. When the address of an unnamed socket is returned, a NULL value is returned for the sun_path field.